### PR TITLE
RC1 CANDIDATE: Domain home in image sample regression fixing, fine-tuning and cleanup

### PR DESCRIPTION
**Pull request by doxiao**
The PR contains the following changes to the domain home in image sample input properties based on inputs from Monica, Tom and Carolyn:
1) change domain home in image sample input property names:
     baseImage ->  domainDomainImageBase
     imagePath -> domainHomeImageBuildPath
2) Make domainHomeImageBase required (no default any more); Operator requires a patched 12.2.1.3 image now.  
3) Change the "image"  property back to be optional, but do not provide the default value. We internally determine the value based on the domainHomeImageBuildPath property.
4) Change the value of domainHomeImageBuildPath to contain the long path under the docker-image GitHub home dir. 
5) Tag the generated image to a generic name that does not contain any WLS release number. For example domain-home-in-image or domain-domain-in-image-wdt. If the "image" property is specified, this step is skipped.

Also include some cleanup in the README and input files based on Tom M's suggestions:
1) Remove references to the "ALWAYS" value for the serverStartupPolicy in README and input files.
2) Add serverStartState: RUNNING to the cluster in domain template YAML.

### CODE REVIEW

File: kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/README.md [hunk 1]
```diff
@@ -71,15 +70,15 @@ The following parameters can be provided in the inputs file.
...
 | `configuredManagedServerCount` | Number of Managed Server instances to generate for the domain. | `2` |
+| `domainHomeImageBase` | WebLogic binary image used to build the WebLogic domain image. The operator requires WebLogic Server 12.2.1.3.0 with patch 28076014 applied. Refer to [Weblogic Docker images](../../../../../site/weblogic-docker-images.md) for details on how to create one. If a different `domainHomeImageBase` (see Configuration table below) is specified, the specified image needs to be built locally or pulled from a repository before the `create-domain.sh` script is executed. | |
+| `domainHomeImageBuildPath` | Location of the WebLogic "domain home in image" Docker image in `https://github.com/oracle/docker-images.git` project. If not specified, use "./docker-images/OracleWebLogic/samples/12213-domain-home-in-image-wdt". Another possible value is "./docker-images/OracleWebLogic/samples/12213-domain-home-in-image" whichuses WLST script, instead of WDT, to generate the domain configuration. | `./docker-images/OracleWebLogic/samples/12213-domain-home-in-image-wdt` |
```
Comment (tbarnes-us) [inline kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/README.md]: whichuses -> which uses a

File: kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/README.md [hunk 2]
```diff
@@ -71,15 +70,15 @@ The following parameters can be provided in the inputs file.
...
 | `exposeAdminNodePort` | Boolean indicating if the Administration Server is exposed outside of the Kubernetes cluster. | `false` |
 | `exposeAdminT3Channel` | Boolean indicating if the T3 administrative channel is exposed outside the Kubernetes cluster. | `false` |
-| `image` | WebLogic Docker image that the domain resource will use. If not specified, the value is the name of the generated Docker image. | `12213-domain-home-in-image-wdt` |
-| `imagePath` | The relative directory of the WebLogic domain home in image Docker image in `https://github.com/oracle/docker-images.git` project under the `docker-images/OracleWebLogic/samples` directory.  | `12213-domain-home-in-image-wdt` |
+| `image` | WebLogic Docker image that the domain resource will pull if needed. You only need to specify this if you are going to push the generated image from the local Docker repository to another Docker repository. If not specified, will use the internally generated image name, say "domain-home-in-image:latest" or "domain-home-in-image-wdt:latest". | |
```
Comment (tbarnes-us) [inline kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/README.md]: If not specified, will use --> If not specified, the sample will use
say --> either

File: kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml [hunk 3]
```diff
@@ -40,9 +40,12 @@ managedServerPort: 8001
 # Boolean indicating if production mode is enabled for the domain
 productionModeEnabled: true
 
-# WebLogic Docker image.
-# The operator requires WebLogic Server 12.2.1.3.0 with patch 28076014 applied. See README.md for more help.
-#image:
+# WebLogic Docker image that the domain resource will pull if needed.
+# You only need to specify this if you are going to push the generated image from the local Docker repository
+# to another Docker repository.
+# If not specified, will use the internally generated image name, say "domain-home-in-image:latest"
```
Comment (tbarnes-us) [inline kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml]: will use --> this sample will use

Comment (tbarnes-us) [reply kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml]: say --> either

File: kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml [hunk 4]
```diff
@@ -85,14 +88,14 @@ namespace: default
...
-# If not specified, use the built-in base image oracle/weblogic:12.2.1.3-developer
-baseImage: oracle/weblogic:12.2.1.3-developer
-
-# The relative directory of the WebLogic domain home in image Docker image in the i
-# `https://github.com/oracle/docker-images.git` project under the `docker-images/OracleWebLogic/samples` directory.
-# If not specified, use 12213-domain-home-in-image-wdt. 
-# Another possible value is 12213-domain-home-in-image which uses WLST script instead of WDT to 
-# generate the domain home.
-imagePath: 12213-domain-home-in-image-wdt
+# WebLogic binary image used to build the WebLogic domain image
```
Comment (tbarnes-us) [inline kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml]: WebLogic binary -> Base WebLogic binary

**Comment by markxnelson** [comment]
@rjeberhard please do not merge this until we have had a chance for people to complete testing on what we already have in 2.0-rc1 domain in image - i don't want it changing out from under them all the time

**Comment by doxiao** [comment]
@markxnelson The PR contains a fix for an new issue that is caused by PR#700. People already started to hit the issue - they are required to set the "image" property, but they don't know what to set. As I mentioned to John in the scrum meeting, it is the baseImage property that requires a patched WLS 12.2.13 image, not the "image", which is usually the image that is generated by the create-domain.sh script. 

We may need to have a separate fix only for the new issue if this PR is not going to be merged soon.

**Comment by markxnelson** [comment]
I understand, and I will merge it when I have tested it

**Comment by simon-meng-cn** [comment]
@mriccell What is the suitable default value for configuredManagedServerCount?

File: kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml [hunk 5]
```diff
@@ -26,7 +26,7 @@ serverStartPolicy: IF_NEEDED
 clusterName: cluster-1
 
 # Number of managed servers to generate for the domain
-configuredManagedServerCount: 2
+configuredManagedServerCount: 5
```
Comment (simon-meng-cn) [inline kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/create-domain-inputs.yaml]: Also need to modify the default value in READ.md

**Comment by doxiao** [comment]
Jenkins is clean except for the known failure in test3 on "Scaling". http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/772.

File: kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/README.md [hunk 6]
```diff
@@ -31,8 +30,13 @@ The script will perform the following steps:
...
 * Create a Kubernetes domain YAML file, `domain.yaml`, in the directory that is created above. This YAML file can be used to create the Kubernetes resource using the `kubectl create -f` or `kubectl apply -f` command.
+```
+kubectl apply -f /path/to/output-directory/weblogic-domains/<domainUID>/domain.yaml
+```
+
+As a convenience, using the `-e` option, the script can optionally create the domain object, which in turn results in the creation of the corresponding WebLogic Server pods and services. This option should only be used in a single node Kubernetes cluster.
 
-As a convenience, using the `-e` option, the script can optionally create the domain object, which in turn results in the creation of the corresponding WebLogic Server pods and services.
+For multi node Kubernetes cluster make sure that the generated image is available on all nodes before creating the domain resource using `kubectl apply -f` command.
```
Comment (rosemarymarano) [inline kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/README.md]: For multi node Kubernetes cluster make sure that the generated image is available on all nodes before creating the domain resource using `kubectl apply -f` command. -> For a multi-node Kubernetes cluster, make sure that the generated image is available on all nodes before creating the domain resource using the `kubectl apply -f` command.

File: kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/README.md [hunk 7]
```diff
@@ -71,15 +75,15 @@ The following parameters can be provided in the inputs file.
...
+| `configuredManagedServerCount` | Number of Managed Server instances to generate for the domain. | `5` |
+| `domainHomeImageBase` | Base WebLogic binary image used to build the WebLogic domain image. The operator requires WebLogic Server 12.2.1.3.0 with patch 28076014 applied. Refer to [Weblogic Docker images](../../../../../site/weblogic-docker-images.md) for details on how to create one. If a different `domainHomeImageBase` (see Configuration table below) is specified, the specified image needs to be built locally or pulled from a repository before the `create-domain.sh` script is executed. | |
+| `domainHomeImageBuildPath` | Location of the WebLogic "domain home in image" Docker image in `https://github.com/oracle/docker-images.git` project. If not specified, use "./docker-images/OracleWebLogic/samples/12213-domain-home-in-image-wdt". Another possible value is "./docker-images/OracleWebLogic/samples/12213-domain-home-in-image" which uses WLST script, instead of WDT, to generate the domain configuration. | `./docker-images/OracleWebLogic/samples/12213-domain-home-in-image-wdt` |
```
Comment (rosemarymarano) [inline kubernetes/samples/scripts/create-weblogic-domain/domain-home-in-image/README.md]: uses WLST script, -> uses a WLST script,

**Comment by markxnelson** [comment]
@markxnelson make sure that you update quickstart.md to match the name of the base image variable

**Comment by doxiao** [comment]
I checked and found no reference to the "baseImage" or "imagePath" in any of the site doc, including site/quickstart.md.

**Comment by rosemarymarano** [comment]
See Step 5b in the quick Start guide, "Copying the sample create-domain-inputs.yaml file and updating your copy with the domainUID (sample-domain1), domain namespace (sample-domain1-ns) and the **base image** (oracle/weblogic:12213-patch-wls-for-k8s).

**Comment by doxiao** [comment]
Sorry, missed it because it did not use the actual property name. fixed.

**Comment by rosemarymarano** [comment]
@doxiao Do quick start guide users need to specify any other properties in the create-domain-inputs.yaml file, in addition to domainUID, domain namespace, and the base image properties)?

**Comment by doxiao** [comment]
Rosemary, that is fine.  I just noticed that the command in the quick start for domain home in image is not correct.
Would you mind if I make a quick fix?

**Pull request merged**